Google cloud functions - Unable to access private npm repositories

- **Expected**
When deploying typescript google cloud functions, private npm repository pacakges are accessible
- **What happens**
Google cloud is not able to access the private packages because the `.npmrc` file with a valid read-only token to be uploaded to the function's repository. 

- Why
This is not available today for pulumi-gcp as ``` codePathOptions.extraIncludePaths not currently supported in GCP.```

- Steps to reproduce:

1. Setup typescript pulumi gcp project
2. Create new http google function referring to a private package:
```
import {parseText} from '@org/private_package';

import * as gcp from '@pulumi/gcp';

let greeting = new gcp.cloudfunctions.HttpCallbackFunction("hello", {
  callback: (req, res) => {
    // Change this code to fit your needs!
    res.send(parseText(`Greetings from ${req.body.name || 'Google Cloud Functions'}!`));
  },
  codePathOptions: {
    extraIncludePaths: ["./.npmrc"],
  },
});

export let url = greeting.httpsTriggerUrl;
```
3. `pulumi up`


- Logs:
```
Diagnostics:
  gcp:cloudfunctions:Function (hello):
    error: Plan apply failed: 1 error occurred:
        * creating urn:pulumi:gcp-functions-test::stackname::gcp:cloudfunctions:CallbackFunction$gcp:cloudfunctions/function:Function::hello: Error waiting for Creating CloudFunctions Function: Error code 3, message: Build failed: {"error": {"canonicalCode": "INVALID_ARGUMENT", "errorMessage": "`gen_package_lock` had stderr output:\nnpm ERR! code E404\nnpm ERR! 404 Not Found: @org/pirvate_repo@^0.0.3\n\nnpm ERR! A complete log of this run can be found in:\nnpm ERR!     /builder/home/.npm/_logs/2019-05-15T17_44_52_086Z-debug.log\n\nerror: `gen_package_lock` returned code: 1", "errorType": "InternalError", "errorId": "F28126B7"}}
 
  pulumi:pulumi:Stack (stackname):
    error: update failed
```

--- a/sdk/nodejs/cloudfunctions/zMixins.ts
+++ b/sdk/nodejs/cloudfunctions/zMixins.ts
@@ -234,9 +234,6 @@ async function computeCodePaths(
     codePathOptions.extraIncludePackages = codePathOptions.extraIncludePackages || [];
     codePathOptions.extraExcludePackages = codePathOptions.extraExcludePackages || [];
 
-    if (codePathOptions.extraIncludePaths.length > 0) {
-        throw new pulumi.ResourceError("codePathOptions.extraIncludePaths not currently supported in GCP.", codePathOptions.logResource);
-    }
     if (codePathOptions.extraIncludePackages.length > 0) {
         throw new pulumi.ResourceError("codePathOptions.extraIncludePackages not currently supported in GCP.", codePathOptions.logResource);
     }
@@ -255,7 +252,9 @@ async function computeCodePaths(
         // Always include the serialized function.
         [serializedFileNameNoExtension + ".js"]: new pulumi.asset.StringAsset(serializedFunction.text),
         // Include package.json and GCP will install packages for us
-        ["package.json"]: new pulumi.asset.StringAsset(packageJson)
+        ["package.json"]: new pulumi.asset.StringAsset(packageJson),
+        // Include files added by user
+        ...processExtraIncludePaths(codePathOptions.extraIncludePaths)
     };
 }
 
@@ -283,6 +282,20 @@ function producePackageJson(excludedPackages: Set<string>): Promise<string> {
             }));
         });
     });
+}
+
+// Create file assets from specified by user list,
+// files will be added to the main directory under the file name thus duplicates will be overridden
+// e.g. ['./dir1/file1', './file1'] will add only the last file.
+function processExtraIncludePaths(extraIncludePaths: string[]): Record<string, pulumi.asset.FileAsset> {
+    const filesToInclude: Record<string, pulumi.asset.FileAsset> = {};
+
+    for (const fullPath of extraIncludePaths){
+        const fileName = filepath.basename(fullPath);
+        filesToInclude[fileName] = new pulumi.asset.FileAsset(fullPath);
+    }
+
+    return filesToInclude;
 }
 
 /**
@@ -341,7 +354,8 @@ export interface CallbackFunctionArgs {
      */
     timeout?: pulumi.Input<number>;
     /**
-     * Boolean variable. Any HTTP request (of a supported type) to the endpoint will trigger function execution. Supported HTTP request types are: POST, PUT, GET, DELETE, and OPTIONS. Endpoint is returned as `https_trigger_url`. Cannot be used with `trigger_bucket` and `trigger_topic`.
+     * Boolean variable. Any HTTP request (of a supported type) to the endpoint will trigger function execution. Supported HTTP request types are: POST, PUT,
+     * GET, DELETE, and OPTIONS. Endpoint is returned as `https_trigger_url`. Cannot be used with `trigger_bucket` and `trigger_topic`.
      */
     triggerHttp?: pulumi.Input<boolean>;
 

